texar/torch/modules/classifiers/bert_classifier.py

```diff
@@ -230,9 +241,14 @@ def forward(self,  # type: ignore
                   ``[batch_size, max_time, num_classes]`` and ``pred`` is of
                   shape ``[batch_size, max_time]``.
         """
-        enc_outputs, pooled_output = self._encoder(inputs,
-                                                   sequence_length,
-                                                   segment_ids)
+        if self.__class__.__name__.startswith('BERT'):
```

Comment:
Let's make this part a separate function instead. Conditioning on class name seems too hacky.